File: SupplierPortal.Web/Modules/ServerTypes/Market/OfferColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OfferRow } from "./OfferRow";

export interface OfferColumns {
    Id: Column<OfferRow>;
    OfferDate: Column<OfferRow>;
    ValueDate: Column<OfferRow>;
}

export class OfferColumns extends ColumnsBase<OfferRow> {
    static readonly columnsKey = 'Market.Offer';
    static readonly Fields = fieldsProxy<OfferColumns>();
}

File: SupplierPortal.Web/Modules/Market/Offer/OfferColumns.cs
using MovieTutorial.MovieDB;
using Serenity.ComponentModel;
using SupplierPortal.Market.Enum;
using System.ComponentModel;

namespace SupplierPortal.Market.Columns;

[ColumnsScript("Market.Offer")]
[BasedOnRow(typeof(OfferRow), CheckNames = true)]
public class OfferColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    [Visible(false)]    
    public int Id { get; set; }
    //[EditLink]
    //public State State { get; set; }
    [EditLink]
    public DateTime OfferDate { get; set; }
    [EditLink]
    public DateTime ValueDate { get; set; }

}